### PR TITLE
Move the v8 snapshot creation to build time

### PR DESCRIPTION
**Pull request by abhirag**
## What problem are you trying to solve?
As of now we create the v8 snapshot lazily during runtime, moving the snapshot creation to build time should improve startup time
## What is your solution?
Move the snapshot creation to build time, the v8 snapshot bytes are then included in the binary
## Alternatives considered
Alternative is the current approach where we create the v8 runtime snapshot lazily during runtime
## What the reviewer should know
The tradeoff is between better startup times and a tad fatter binary. I can try and profile to get some concrete numbers if the tradeoff seems unacceptable. The approach I put forward is how Deno's cli seems to be handling snapshotting too. Also I had to fix some clippy lints ([manual_retain](https://rust-lang.github.io/rust-clippy/master/index.html#/manual_retain), [iter_overeager_cloned](https://rust-lang.github.io/rust-clippy/master/index.html#/iter_overeager_cloned))

### CODE REVIEW

**Comment by juli1** [comment]
Sorry for the late reply, just looked at the project and noticed the PR! We will get back to you shortly on this!

File: kernel/build.rs [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+            snapshot_path,
+            startup_snapshot: None,
+            skip_op_registration: false,
+            extensions: vec![deno_extension::init_ops_and_esm()],
+            compression_cb: None,
+            with_runtime_cb: None,
+        },
+    );
+}
+
```
Comment (juli1) [inline kernel/build.rs:24]: Please put a comment explaining what it does. Especially that it build the snapshot that will be later used by a static variable. We may also put a reference to the cargo book about the environment variable, explaining where to put [references to `OUT_DIR`](https://doc.rust-lang.org/cargo/reference/environment-variables.html)

File: kernel/src/analysis/javascript.rs [hunk 2]
```diff
@@ -3,25 +3,17 @@ use crate::model::analysis::{
...
 
-lazy_static! {
-    static ref STARTUP_DATA: Vec<u8> = {
-        let code: FastString = FastString::from_static(include_str!("./js/stella.js"));
-        let mut rt = JsRuntimeForSnapshot::new(RuntimeOptions::default());
-        rt.execute_script("common_js", code).unwrap();
-        rt.snapshot().to_vec()
-    };
-}
+static STARTUP_DATA: &[u8] = include_bytes!(concat!(env!("OUT_DIR"), "/DENO_SNAPSHOT.bin"));
```
Comment (juli1) [inline kernel/src/analysis/javascript.rs:16]: Put a comment that explains this file is built at compile-time in `build.rs`

**Review by juli1 — CHANGES_REQUESTED** [review]

**Review by juli1 — APPROVED** [review]

**Comment by abhirag** [comment]
@juli1 Thanks for your feedback, added the required comments

**Pull request merged**